core/src/logging.jl

```diff
@@ -38,13 +38,21 @@ function log_bottlenecks(model; converged::Bool)
         flow_error = @. abs(cache.nlsolver.cache.atmp / u)
         perm = sortperm(flow_error; rev = true)
         errors = Pair{Symbol, Float64}[]
+        error_count = 0
+        max_errors = 5
         for i in perm
             node_id = Symbol(id_from_state_index(p, u, i))
             error = flow_error[i]
-            if error < model.config.solver.reltol
+            # Stop reporting errors if they are too small or too many
+            if error < model.config.solver.reltol || error_count >= max_errors
                 break
             end
-            push!(errors, node_id => error)
+            if isnan(error)
+                continue
+            else
+                push!(errors, node_id => error)
+                error_count += 1
+            end
```

Comment:
```julia
errors = Pair{Symbol, Float64}[]
error_count = 0
max_errors = 5
# Iterate over the errors in descending order
for i in sortperm(flow_error; rev = true)
    node_id = Symbol(id_from_state_index(p, u, i))
    error = flow_error[i]
    isnan(error) && continue  # nan are sorted as largest
    # Stop reporting errors if they are too small or too many
    if error < model.config.solver.reltol || error_count >= max_errors
        break
    end
    push!(errors, node_id => error)
    error_count += 1
```

That makes it more readable for me.

Reply:
Maybe max errors should not be set in the function but as a constant on top of the file.